p2p/chain.py

```diff
@@ -403,7 +404,9 @@ def _handle_block_headers(self, headers: List[BlockHeader]) -> None:
         # the executor when the list() is applied to it.
         receipts_tries = await self.wait(loop.run_in_executor(self._executor, list, iterator))
         downloaded: List[DownloadedBlockPart] = []
-        for (receipts, (receipt_root, trie_dict_data)) in zip(receipts_by_block, receipts_tries):
+        # TODO: figure out why mypy is losing the type of the receipts_tries
+        # so we can get rid of the ignore
```

Comment:
I'm not sure what this means. Shouldn't mypy treat receipts_tries's type as Any since that's what self.wait() returns?

Reply:
[`wait(...)`](https://github.com/cburgdorf/py-evm/blob/69892c3ff720ad437d76ca29d1c94df81c5380e1/p2p/service.py#L45-L52) is generic now and should return the right type which works in principle but fails at this place.